sklearn/model_selection/_validation.py

```diff
@@ -727,12 +727,35 @@ def _fit_and_predict(estimator, X, y, train, test, verbose, fit_params,
     predictions = func(X_test)
     if method in ['decision_function', 'predict_proba', 'predict_log_proba']:
         n_classes = len(set(y))
-        predictions_ = np.zeros((_num_samples(X_test), n_classes))
-        if method == 'decision_function' and len(estimator.classes_) == 2:
-            predictions_[:, estimator.classes_[-1]] = predictions
-        else:
-            predictions_[:, estimator.classes_] = predictions
-        predictions = predictions_
+        if n_classes != len(estimator.classes_):
+            if predictions.ndim == 2 and \
+                    predictions.shape[1] != len(estimator.classes_):
+                # This handles the case when the shape of predictions
+                # does not match the number of classes used to train
+                # it with. This case is found when sklearn.svm.SVC is
+                # set to `decision_function_shape='ovo'`.
+                raise ValueError('Output shape (-1, {}) of {} does not '
```

Comment:
It's supposed to mean that only the second value in the shape matters. I initially just put in the shape but the test was failing because `.format()` would put in either `(10, 10)` or `(10L, 10L)`.

 Do you think something like "Inner dimension of output shape {}" would be better?

Reply:
That's what I half-guessed but this is very confusing I am afraid. 

> I initially just put in the shape but the test was failing because .format() would put in either (10, 10) or (10L, 10L).

So you made a worse error message just to get the test to pass. I think you need to set your priorities straight ;-) ! You can use assert_raises_regex with a regex. Use git `grep L?` for an example. The important thing is to have a good error message.

Reply:
Ah, my bad. I didn't know about assert_raises_regex. Will get on a fix immediately!